ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+eval "$(dev-env/bin/dade assist)"
+
+readonly BUF_IMAGE_TMPDIR="$(mktemp -d)"
+trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
+
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+if [[ "${CURRENT_BRANCH}" == "main" || "${SYSTEM_PULLREQUEST_TARGETBRANCH}" == "main" ]]; then
+  GIT_TAG_SCOPE="--no-merged" # Release tags are not reachable from `main` because release branches predate them
+else
+  GIT_TAG_SCOPE="--merged"    # The relevant most recent release tag is always reachable from its release branch
```

Comment:
This comment is _mostly_ true. In the current situation, for example, this would return `1.16.0` for the current 1.17.x branch, regardless of whether we ever had a 1.16.1 (right now we don't, but it wouldn't be returned if we did).

Reply:
Let me know if dbc9267 solves this.